### ISSUE TITLE
 forward() got an unexpected keyword

### ISSUE THREAD

**Issue by wwongkamjan** [facebookresearch/diplomacy_searchbot#9]
Greetings

I have been trying to retrain your diplomacy_searchbot. However, I could not run this line which is to compare 1vs 6

```
python run.py --adhoc --cfg conf/c01_ag_cmp/cmp.prototxt \
    I.agent_one=agents/dipnet_20200827_iclr_v_humans.prototxt \
    I.agent_six=agents/searchbot_neurips21_human_dnvi_npu.prototxt
```

The full error log is here
```
I0826 18:48:35 [run:147] Config: conf/c01_ag_cmp/cmp.prototxt
I0826 18:48:35 [run:148] Overrides: ['I.agent_one=agents/dipnet_20200827_iclr_v_humans.prototxt', 'I.agent_six=agents/searchbot_neurips21_human_dnvi_npu.prototxt']
I0826 18:48:35 [run:156] Exp dir: /home/wwongkam/results/diplomacy/adhoc/2022-08-26T184835.381498/c01_ag_cmp/cmp/I.age@agents_dipnet_20200827_iclr_v_humans.prototxt@I.age@agents_searchbot_neurips21_human_dnvi_npu._56704896
I0826 18:48:35 [run:157] Job status [before run]: Status.NOT_STARTED
I0826 18:48:35 [run:83] Cwd: /home/wwongkam/diplomacy_searchbot
I0826 18:48:35 [run:84] Task: compare_agents
I0826 18:48:35 [run:85] Cfg:
agent_one {
  model_sampled {
    model_path: "blueprint.pt"
    temperature: 0.5
  }
}
agent_six {
  searchbot {
    model_path: "models/neurips21_human_dnvi_npu_epoch000500.ckpt"
    n_rollouts: 512
    value_model_path: "models/neurips21_human_dnvi_npu_value_epoch000500.ckpt"
    cache_rollout_results: true
    loser_bp_value: 0.019999999552965164
    loser_bp_iter: 64.0
    rollouts_cfg {
      n_threads: 56
      temperature: 0.75
      top_p: 0.949999988079071
      max_rollout_length: 0
      average_n_rollouts: 1
    }
    plausible_orders_cfg {
      n_plausible_orders: 30
      max_actions_units_ratio: 3.5
      req_size: 1024
    }
  }
}
power_one: AUSTRIA
out: "output.json"
seed: 0
draw_on_stalemate_years: 5
launcher {
  local {
    use_local: true
  }
}

I0826 18:48:35 [util:222] Git revision: ed2966d285ae828720b7b817bb42c617da194b9f
I0826 18:48:35 [util:238] Found unsubmitted diff. Saving to /home/wwongkam/diplomacy_searchbot/workdir.diff
I0826 18:48:35 [run:88] Is on slurm: False
I0826 18:48:35 [run:89] Job env: JobEnvironment(job_id=1, hostname=MSI, local_rank=0(1), node=0(1), global_rank=0(1))
I0826 18:48:35 [run:92] Is master: True
I0826 18:48:35 [run:34] Set seed to 0
I0826 18:48:40 [searchbot_agent:212] Initialized SearchBotAgent: {'max_batch_size': 700, 'model': <fairdiplomacy.agents.model_wrapper.ModelWrapper object at 0x7fe1f7286810>, 'model_rollouts': <fairdiplomacy.agents.model_rollouts.ModelRollouts object at 0x7fe1f7286ad0>, 'n_rollouts': 512, 'cache_rollout_results': True, 'precompute_cache': False, 'enable_compute_nash_conv': False, 'n_plausible_orders': 30, 'use_optimistic_cfr': True, 'use_final_iter': True, 'use_pruning': False, 'bp_iters': 0, 'bp_prob': 0.0, 'loser_bp_iter': 64.0, 'loser_bp_value': 0.019999999552965164, 'share_strategy': False, 'reset_seed_on_rollout': False, 'max_seconds': 0, 'order_sampler': <fairdiplomacy.agents.plausible_order_sampling.PlausibleOrderSampler object at 0x7fe1f7269ad0>, 'order_aug_cfg': }
I0826 18:48:40 [env:100] Starting order prediction for turn SPRING 1901 MOVEMENT
Traceback (most recent call last):
  File "run.py", line 102, in <module>
    heyhi.parse_args_and_maybe_launch(main)
  File "/home/wwongkam/diplomacy_searchbot/heyhi/run.py", line 114, in parse_args_and_maybe_launch
    maybe_launch(main, exp_root=get_exp_dir(PROJECT_NAME), overrides=overrides, **kwargs)
  File "/home/wwongkam/diplomacy_searchbot/heyhi/run.py", line 166, in maybe_launch
    util.run_with_config(main, exp_handle, cfg, overrides, ckpt_dir, log_level)
  File "/home/wwongkam/diplomacy_searchbot/heyhi/util.py", line 588, in run_with_config
    callable()
  File "/home/wwongkam/diplomacy_searchbot/heyhi/util.py", line 371, in wrapped
    result = f(*args, **kwargs)
  File "run.py", line 98, in main
    return TASKS[task](cfg)
  File "run.py", line 53, in compare_agents
    result = run_1v6_trial(agent_one, agent_six, power_string, cfg, cf_agent=cf_agent)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/compare_agents.py", line 63, in run_1v6_trial
    scores = env.process_all_turns(max_turns=cfg.max_turns)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/env.py", line 135, in process_all_turns
    self.process_turn()
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/env.py", line 101, in process_turn
    power_orders = self.policy_profile.get_all_power_orders(self.game)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/env.py", line 49, in get_all_power_orders
    game, self._agent_one_power
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/agents/model_sampled_agent.py", line 32, in get_orders
    return self.get_orders_many_powers(game, [power], **kwargs)[power]
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/agents/model_sampled_agent.py", line 45, in get_orders_many_powers
    actions, _, _ = self.model.do_model_request(inputs, temperature=temperature, top_p=top_p)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/agents/model_wrapper.py", line 99, in do_model_request
    device=self.device,
  File "/home/wwongkam/miniconda3/lib/python3.7/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/utils/batching.py", line 21, in batched_forward
    batch_results = callable(batch_data)
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/agents/model_wrapper.py", line 95, in <lambda>
    batch_repeat_interleave=batch_repeat_interleave,
  File "/home/wwongkam/diplomacy_searchbot/fairdiplomacy/agents/model_wrapper.py", line 157, in _forward_policy
    y = self.model(**x, need_value=False, pad_to_max=True)
  File "/home/wwongkam/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
TypeError: forward() got an unexpected keyword argument 'x_scoring_system'
```

I am having trouble finding out where is the root cause of this error. May I ask for your time to look into this issue, please?

Thank you so much

**Comment by akhti** [comment]
Hey!

Pushed fix [3f93a03](https://github.com/facebookresearch/diplomacy_searchbot/commit/3f93a03a38d9cf0ccc062f3847a278d3808a2032)

Please try the following

```
git pull
make compile
```

After that things should work

**Comment by wwongkamjan** [comment]
Hi 

Now it works! Thank you so much.
